Plugins/Json/include/Acts/Plugins/Json/GridJsonConverter.hpp

```diff
@@ -83,30 +144,29 @@ nlohmann::json toJson(const grid_type& grid) {
 /// @note the axis generator also defines the grid dimension
 ///
 /// @return a grid object
-template <typename axis_generator_type>
+template <typename axis_generator_type,
+          typename value_type = std::vector<std::size_t>>
 auto fromJson(const nlohmann::json& jGrid,
               const axis_generator_type& aGenerator) {
```

Comment:
Why not just pass the grid rather than the generator?

Reply:
We use everywhere the following pattern:
```Object fromJson(const nlohmann::json& jInput, const Options& a)```

I can keep this pattern, if the Options is the way how to generate the grid, otherwise I would have to change it to

```void fromJson(const nlohmann::json& jinput, grid_type& grid)```

Where the grid is filled in the `fromJson` method - which I don't like.

Reply:
@paulgessinger does this satisfy you as an answer?

Reply:
Not sure I get it. You're not passing an options struct here, but the generator object. What's the difference between passing the generator object that you call to make the grid, compare to just passing the grid?

Reply:
The grid is the object that is created by the `fromJson` call. Everywhere in the code we have the same signature and code usage:

```Object fromJson(const nlohmann::json& jInput, const Options& a)```

Now, in case of the Grid, the return object can have different type, so I need some sort of template mechanism. I can not do 

```template <typename GridType> GridType fromJson(const nlohmann::json& jInput)``` because there's neither default nor standard constructors for Grid objects.

Hence, the only other way to do it is:

```template <typename GridType> void fillFromJson(const nlohmann::json& jInput, GridType& grid)```

I, personally, like the generator more, because this code already exists, and I can reuse it - and, as I said before - it keeps the same/similar message signature as in most of the other json code.